--- a/libretro.c	2018-04-14 01:45:03.000000000 +0200
+++ b/libretro.c	2018-08-04 09:40:11.683031034 +0200
@@ -122,37 +122,37 @@
 
    if(strcmp(extension, ".dsk") == 0){
       if (is_auto){
-         strcpy(msx_type, "MSX2+");
+         strcpy(msx_type, "MSX2");
       }
       return MEDIA_TYPE_DISK;
    }
    else if(strcmp(extension, ".m3u") == 0){
       if (is_auto){
-         strcpy(msx_type, "MSX2+");
+         strcpy(msx_type, "MSX2");
       }
       return MEDIA_TYPE_DISK_BUNDLE;
    }
    else if(strcmp(extension, ".cas") == 0){
       if (is_auto){
-         strcpy(msx_type, "MSX2+");
+         strcpy(msx_type, "MSX2");
       }
       return MEDIA_TYPE_TAPE;
    }
    else if(strcmp(extension, ".rom") == 0){
       if (is_auto){
-         strcpy(msx_type, "MSX2+");
+         strcpy(msx_type, "MSX2");
       }
       return MEDIA_TYPE_CART;
    }
    else if(strcmp(extension, ".mx1") == 0){
       if (is_auto){
-         strcpy(msx_type, "MSX2+");
+         strcpy(msx_type, "MSX2");
       }
       return MEDIA_TYPE_CART;
    }
    else if(strcmp(extension, ".mx2") == 0){
       if (is_auto){
-         strcpy(msx_type, "MSX2+");
+         strcpy(msx_type, "MSX2");
       }
       return MEDIA_TYPE_CART;
    }
@@ -574,7 +574,7 @@
    static const struct retro_variable vars[] = {
       { "bluemsx_msxtype", "Machine Type (Restart); Auto|MSX|MSXturboR|MSX2|MSX2+|SEGA - SG-1000|SEGA - SC-3000|SEGA - SF-7000|SVI - Spectravideo SVI-318|SVI - Spectravideo SVI-328|SVI - Spectravideo SVI-328 MK2|ColecoVision|Coleco (Spectravideo SVI-603)" },
       { "bluemsx_overscan", "Crop Overscan; disabled|enabled|MSX2" },
-      { "bluemsx_vdp_synctype", "VDP Sync Type (Restart); Auto|50Hz|60Hz" },
+      { "bluemsx_vdp_synctype", "VDP Sync Type (Restart); 60Hz|50Hz|Auto" },
       { "bluemsx_nospritelimits", "No Sprite Limit; OFF|ON" },
       { "bluemsx_ym2413_enable", "Sound YM2413 Enable (Restart); enabled|disabled" },
       { "bluemsx_cartmapper", "Cart Mapper Type (Restart); Auto|Normal|mirrored|basic|0x4000|0xC000|ascii8|ascii8sram|ascii16|ascii16sram|ascii16nf|konami4|konami4nf|konami5|konamisynth|korean80|korean90|korean126|MegaFlashRomScc|MegaFlashRomSccPlus|msxdos2|scc|sccexpanded|sccmirrored|sccplus|snatcher|sdsnatcher|SegaBasic|SG1000|SG1000Castle|SG1000RamA|SG1000RamB|SC3000" },
