diff --git a/Src/Input/InputEvent.h b/Src/Input/InputEvent.h
index 5f3e474..23647b3 100755
--- a/Src/Input/InputEvent.h
+++ b/Src/Input/InputEvent.h
@@ -154,25 +154,21 @@ const char* inputEventCodeToString(int eventCode);
 #define EC_JOY1_RIGHT   103
 #define EC_JOY1_BUTTON1 104
 #define EC_JOY1_BUTTON2 105
-#define EC_JOY1_BUTTON3 106
-#define EC_JOY1_BUTTON4 107
-#define EC_JOY1_BUTTON5 108
-#define EC_JOY1_BUTTON6 109
-#define EC_JOY1_WHEELA  136
-#define EC_JOY1_WHEELB  137
+#define EC_F1           106
+#define EC_F2           107
+#define EC_F3           108
+#define EC_F4           109
+#define EC_F5           110
 
-#define EC_JOY2_UP      110
-#define EC_JOY2_DOWN    111
-#define EC_JOY2_LEFT    112
-#define EC_JOY2_RIGHT   113
-#define EC_JOY2_BUTTON1 114
-#define EC_JOY2_BUTTON2 115
-#define EC_JOY2_BUTTON3 116
-#define EC_JOY2_BUTTON4 117
-#define EC_JOY2_BUTTON5 118
-#define EC_JOY2_BUTTON6 119
-#define EC_JOY2_WHEELA  138
-#define EC_JOY2_WHEELB  139
+#define EC_JOY2_UP      111
+#define EC_JOY2_DOWN    112
+#define EC_JOY2_LEFT    113
+#define EC_JOY2_RIGHT   114
+#define EC_JOY2_BUTTON1 115
+#define EC_JOY2_BUTTON2 116
+#define EC_JOY2_BUTTON3 117
+#define EC_JOY2_BUTTON4 118
+#define EC_JOY2_BUTTON5 119
 
 #define EC_COLECO1_0    120
 #define EC_COLECO1_1    121
@@ -241,6 +237,15 @@ const char* inputEventCodeToString(int eventCode);
 
 #define EC_KEYCOUNT                     198
 
+#define EC_JOY1_BUTTON3 200
+#define EC_JOY1_BUTTON4 201
+#define EC_JOY1_BUTTON5 202
+#define EC_JOY1_BUTTON6 203
+#define EC_JOY1_WHEELA  204
+#define EC_JOY1_WHEELB  205
+#define EC_JOY2_BUTTON6 206
+#define EC_JOY2_WHEELA  207
+#define EC_JOY2_WHEELB  208
 // Inlines
 extern int eventMap[256];
 
diff --git a/libretro.c b/libretro.c
index b6adadf..4f95a3d 100644
--- a/libretro.c
+++ b/libretro.c
@@ -122,37 +122,37 @@ int get_media_type(const char* filename)
 
    if(strcmp(extension, ".dsk") == 0){
       if (is_auto){
-         strcpy(msx_type, "MSX2+");
+         strcpy(msx_type, "MSX2");
       }
       return MEDIA_TYPE_DISK;
    }
    else if(strcmp(extension, ".m3u") == 0){
       if (is_auto){
-         strcpy(msx_type, "MSX2+");
+         strcpy(msx_type, "MSX2");
       }
       return MEDIA_TYPE_DISK_BUNDLE;
    }
    else if(strcmp(extension, ".cas") == 0){
       if (is_auto){
-         strcpy(msx_type, "MSX2+");
+         strcpy(msx_type, "MSX2");
       }
       return MEDIA_TYPE_TAPE;
    }
    else if(strcmp(extension, ".rom") == 0){
       if (is_auto){
-         strcpy(msx_type, "MSX2+");
+         strcpy(msx_type, "MSX2");
       }
       return MEDIA_TYPE_CART;
    }
    else if(strcmp(extension, ".mx1") == 0){
       if (is_auto){
-         strcpy(msx_type, "MSX2+");
+         strcpy(msx_type, "MSX2");
       }
       return MEDIA_TYPE_CART;
    }
    else if(strcmp(extension, ".mx2") == 0){
       if (is_auto){
-         strcpy(msx_type, "MSX2+");
+         strcpy(msx_type, "MSX2");
       }
       return MEDIA_TYPE_CART;
    }
@@ -422,21 +422,21 @@ static unsigned btn_map[EC_KEYCOUNT] =
    RETRO_DEVICE_ID_JOYPAD_RIGHT,    //EC_JOY1_RIGHT   103
    RETRO_DEVICE_ID_JOYPAD_A,        //EC_JOY1_BUTTON1 104
    RETRO_DEVICE_ID_JOYPAD_B,        //EC_JOY1_BUTTON2 105
-   RETRO_DEVICE_ID_JOYPAD_Y,        //EC_JOY1_BUTTON3 106
-   RETRO_DEVICE_ID_JOYPAD_X,        //EC_JOY1_BUTTON4 107
-   RETRO_DEVICE_ID_JOYPAD_START,    //EC_JOY1_BUTTON5 108
-   RETRO_DEVICE_ID_JOYPAD_SELECT,   //EC_JOY1_BUTTON6 109
-
-   RETRO_DEVICE_ID_JOYPAD_UP,       //EC_JOY2_UP      110
-   RETRO_DEVICE_ID_JOYPAD_DOWN,     //EC_JOY2_DOWN    111
-   RETRO_DEVICE_ID_JOYPAD_LEFT,     //EC_JOY2_LEFT    112
-   RETRO_DEVICE_ID_JOYPAD_RIGHT,    //EC_JOY2_RIGHT   113
-   RETRO_DEVICE_ID_JOYPAD_A,        //EC_JOY2_BUTTON1 114
-   RETRO_DEVICE_ID_JOYPAD_B,        //EC_JOY2_BUTTON2 115
-   RETRO_DEVICE_ID_JOYPAD_Y,        //EC_JOY2_BUTTON3 116
-   RETRO_DEVICE_ID_JOYPAD_X,        //EC_JOY2_BUTTON4 117
-   RETRO_DEVICE_ID_JOYPAD_START,    //EC_JOY2_BUTTON5 118
-   RETRO_DEVICE_ID_JOYPAD_SELECT,   //EC_JOY2_BUTTON6 119
+   RETRO_DEVICE_ID_JOYPAD_START,    //EC_F1           106
+   RETRO_DEVICE_ID_JOYPAD_L,        //EC_F2           107
+   RETRO_DEVICE_ID_JOYPAD_R,        //EC_F3           108
+   RETRO_DEVICE_ID_JOYPAD_X,        //EC_F4           109
+   RETRO_DEVICE_ID_JOYPAD_SELECT,   //EC_F5           110
+
+   RETRO_DEVICE_ID_JOYPAD_UP,       //EC_JOY2_UP      111
+   RETRO_DEVICE_ID_JOYPAD_DOWN,     //EC_JOY2_DOWN    112
+   RETRO_DEVICE_ID_JOYPAD_LEFT,     //EC_JOY2_LEFT    113
+   RETRO_DEVICE_ID_JOYPAD_RIGHT,    //EC_JOY2_RIGHT   114
+   RETRO_DEVICE_ID_JOYPAD_A,        //EC_JOY2_BUTTON1 115
+   RETRO_DEVICE_ID_JOYPAD_B,        //EC_JOY2_BUTTON2 116
+   RETRO_DEVICE_ID_JOYPAD_L,        //EC_JOY2_BUTTON3 117
+   RETRO_DEVICE_ID_JOYPAD_R,        //EC_JOY2_BUTTON4 118
+   RETRO_DEVICE_ID_JOYPAD_X,        //EC_JOY2_BUTTON5 119
 
    RETROK_UNKNOWN,                  //EC_COLECO1_0    120
    RETROK_UNKNOWN,                  //EC_COLECO1_1    121
@@ -571,7 +571,7 @@ void retro_set_environment(retro_environment_t cb)
    static const struct retro_variable vars[] = {
       { "bluemsx_msxtype", "Machine Type (Restart); Auto|MSX|MSXturboR|MSX2|MSX2+|SEGA - SG-1000|SEGA - SC-3000|SEGA - SF-7000|SVI - Spectravideo SVI-318|SVI - Spectravideo SVI-328|SVI - Spectravideo SVI-328 MK2|ColecoVision|Coleco (Spectravideo SVI-603)" },
       { "bluemsx_overscan", "Crop Overscan; disabled|enabled|MSX2" },
-      { "bluemsx_vdp_synctype", "VDP Sync Type (Restart); Auto|50Hz|60Hz" },
+      { "bluemsx_vdp_synctype", "VDP Sync Type (Restart); 60Hz|50Hz|Auto" },
       { "bluemsx_nospritelimits", "No Sprite Limit; OFF|ON" },
       { "bluemsx_ym2413_enable", "Sound YM2413 Enable (Restart); enabled|disabled" },
       { "bluemsx_cartmapper", "Cart Mapper Type (Restart); Auto|Normal|mirrored|basic|0x4000|0xC000|ascii8|ascii8sram|ascii16|ascii16sram|ascii16nf|konami4|konami4nf|konami5|konamisynth|korean80|korean90|korean126|MegaFlashRomScc|MegaFlashRomSccPlus|msxdos2|scc|sccexpanded|sccmirrored|sccplus|snatcher|sdsnatcher|SegaBasic|SG1000|SG1000Castle|SG1000RamA|SG1000RamB|SC3000" },
