--- a/src/joystick_driver.c	2019-05-11 17:32:24.000000000 +0200
+++ b/src/joystick_driver.c	2019-05-24 18:01:43.963900642 +0200
@@ -52,6 +52,8 @@
 int g_joystick_y2_axis = 3;                             // SDL2
 int g_joystick_button_2 = 2;                    // SDL2
 int g_joystick_button_3 = 3;            // SDL2
+int g_joystick_button_hk = 4;                    // SDL2
+int g_joystick_button_start = 5;            // SDL2
 #define JOY2SUPPORT
 
 
@@ -114,6 +116,10 @@
     } else {
       g_paddle_buttons = g_paddle_buttons & (~8);
     }
+    if (SDL_JoystickGetButton(gGameController, g_joystick_button_hk) && SDL_JoystickGetButton(gGameController, g_joystick_button_start)) {
+      SDL_Event event = { SDL_QUIT };
+      SDL_PushEvent(&event);
+    }
     paddle_update_trigger_dcycs(dcycs);
   }
 }
--- a/src/options.c	2019-05-11 17:32:24.000000000 +0200
+++ b/src/options.c	2019-05-24 17:56:02.385012030 +0200
@@ -79,6 +79,8 @@
 extern int g_joystick_button_1;       // defined in joystick_driver.c
 extern int g_joystick_button_2;       // defined in joystick_driver.c
 extern int g_joystick_button_3;       // defined in joystick_driver.c
+extern int g_joystick_button_hk;      // defined in joystick_driver.c
+extern int g_joystick_button_start;   // defined in joystick_driver.c
 
 
 // DEPRECATED: force bit depth (15/16/24) for X-Windows, might still work.
@@ -304,6 +306,24 @@
         glogf("%s Setting joystick Button 3 to Gamepad %d", parse_log_prefix, tmp1);
         g_joystick_button_3 = tmp1;
         i++;
+    } else if(!strcmp("-joy-bhk", argv[i])) {
+        if((i+1) >= argc) {
+          glogf("%s Error, option '-joy-bhk' missing argument", parse_log_prefix);
+          exit(1);
+        }
+        tmp1 = strtol(argv[i+1], 0, 0); // no bounds check, not sure what ids we get
+        glogf("%s Setting joystick Button hk to Gamepad %d", parse_log_prefix, tmp1);
+        g_joystick_button_hk = tmp1;
+        i++;
+    } else if(!strcmp("-joy-bstart", argv[i])) {
+        if((i+1) >= argc) {
+          glogf("%s Error, option '-joy-bstart' missing argument", parse_log_prefix);
+          exit(1);
+        }
+        tmp1 = strtol(argv[i+1], 0, 0); // no bounds check, not sure what ids we get
+        glogf("%s Setting joystick Button start to Gamepad %d", parse_log_prefix, tmp1);
+        g_joystick_button_start = tmp1;
+        i++;
     } else if(!strcmp("-dhr140", argv[i])) {
       glogf("%s Using simple dhires color map", parse_log_prefix);
       g_use_dhr140 = 1;
