--- a/frontend/libpicofe/in_sdl.c	2019-03-25 22:48:57.000000000 +0100
+++ b/frontend/libpicofe/in_sdl.c	2019-07-04 22:27:21.015760088 +0200
@@ -175,16 +175,6 @@
 	if (pdata->key_names)
 		key_names = pdata->key_names;
 
-	state = calloc(1, sizeof(*state));
-	if (state == NULL) {
-		fprintf(stderr, "in_sdl: OOM\n");
-		return;
-	}
-
-	state->drv = drv;
-	in_register(IN_SDL_PREFIX "keys", -1, state, SDLK_LAST,
-		key_names, 0);
-
 	/* joysticks go here too */
 	SDL_InitSubSystem(SDL_INIT_JOYSTICK);
 
@@ -207,7 +197,17 @@
 		in_register(name, -1, state, SDLK_LAST, key_names, 0);
 	}
 
-	if (joycount > 0)
+  state = calloc(1, sizeof(*state));
+  if (state == NULL) {
+    fprintf(stderr, "in_sdl: OOM\n");
+    return;
+  }
+
+  state->drv = drv;
+  in_register(IN_SDL_PREFIX "keys", -1, state, SDLK_LAST,
+              key_names, 0);
+
+  if (joycount > 0)
 		SDL_JoystickEventState(SDL_ENABLE);
 }
 
