--- a/frontend/plugin_lib.c	2019-03-25 22:48:57.000000000 +0100
+++ b/frontend/plugin_lib.c	2019-07-08 09:18:22.937310685 +0200
@@ -16,6 +16,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <pthread.h>
+#include <SDL_events.h>
 
 #include "libpicofe/fonts.h"
 #include "libpicofe/input.h"
@@ -612,6 +613,53 @@
 	emu_set_action(emu_act);
 
 	in_keystate = actions[IN_BINDTYPE_PLAYER12];
+
+	if (in_keystate & (1u << DKEY_SELECT))
+  {
+    if (in_keystate & (1u << DKEY_START))
+    {
+      printf("Quit requested\n");
+      SDL_Event quit;
+      quit.type = SDL_QUIT;
+      SDL_PushEvent(&quit);
+    }
+    else if (in_keystate & (1u << DKEY_TRIANGLE))
+    {
+      emu_set_action(SACTION_SAVE_STATE);
+    }
+    else if (in_keystate & (1u << DKEY_SQUARE))
+    {
+      emu_set_action(SACTION_LOAD_STATE);
+    }
+    else if (in_keystate & (1u << DKEY_DOWN))
+    {
+      emu_set_action(SACTION_NEXT_SSLOT);
+    }
+    else if (in_keystate & (1u << DKEY_UP))
+    {
+      emu_set_action(SACTION_PREV_SSLOT);
+    }
+    else if (in_keystate & (1u << DKEY_RIGHT))
+    {
+      emu_set_action(SACTION_FAST_FORWARD);
+    }
+    else if (in_keystate & (1u << DKEY_CROSS))
+    {
+      emu_set_action(SACTION_ENTER_MENU);
+    }
+    else if (in_keystate & (1u << DKEY_CIRCLE))
+    {
+      emu_set_action(SACTION_TOGGLE_FPS);
+    }
+    else if (in_keystate & (1u << DKEY_L1))
+    {
+      emu_set_action(SACTION_TOGGLE_FSKIP);
+    }
+    else if (in_keystate & (1u << DKEY_R1))
+    {
+      emu_set_action(SACTION_SCREENSHOT);
+    }
+  }
 }
 #else /* MAEMO */
 extern void update_input(void);
