diff --git a/toolchain/toolchain-common.in b/toolchain/toolchain-common.in
index 6c285992eb..050f79d793 100644
--- a/toolchain/toolchain-common.in
+++ b/toolchain/toolchain-common.in
@@ -254,10 +254,15 @@ config BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_13
 	bool
 	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_12
 
+config BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_14
+	bool
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_13
+
 # This order guarantees that the highest version is set, as kconfig
 # stops affecting a value on the first matching default.
 config BR2_TOOLCHAIN_HEADERS_AT_LEAST
 	string
+	default "4.14" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_14
 	default "4.13" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_13
 	default "4.12" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_12
 	default "4.11" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_11
