diff --git a/package/sqlite/sqlite.mk b/package/sqlite/sqlite.mk
index 0a5c1bb8f9..ef4843753e 100644
--- a/package/sqlite/sqlite.mk
+++ b/package/sqlite/sqlite.mk
@@ -6,7 +6,7 @@
 
 SQLITE_VERSION = 3150000
 SQLITE_SOURCE = sqlite-autoconf-$(SQLITE_VERSION).tar.gz
-SQLITE_SITE = http://www.sqlite.org/2016
+SQLITE_SITE = http://www.sqlite.org/2015
 SQLITE_LICENSE = Public domain
 SQLITE_LICENSE_FILES = tea/license.terms
 SQLITE_INSTALL_STAGING = YES
