diff --git a/package/kodi-visualisation-spectrum/Config.in b/package/kodi-visualisation-spectrum/Config.in
index 3b2b0a25e2..bc5283a660 100644
--- a/package/kodi-visualisation-spectrum/Config.in
+++ b/package/kodi-visualisation-spectrum/Config.in
@@ -1,4 +1,5 @@
 config BR2_PACKAGE_KODI_VISUALISATION_SPECTRUM
+    depends on BR2_PACKAGE_KODI
 	bool "kodi-visualisation-spectrum"
 	help
 	  Spectrum visualiser for Kodi
