diff --git a/package/bluez_utils/bluez_utils.mk b/package/bluez_utils/bluez_utils.mk
index 1c93f2e5f5..5e37b81cd2 100644
--- a/package/bluez_utils/bluez_utils.mk
+++ b/package/bluez_utils/bluez_utils.mk
@@ -27,7 +27,8 @@ endif
 ifeq ($(BR2_PACKAGE_BLUEZ_UTILS_AUDIO),y)
 BLUEZ_UTILS_DEPENDENCIES +=	\
 	alsa-lib		\
-	libsndfile
+	libsndfile		\
+	libvorbis
 BLUEZ_UTILS_CONF_OPTS +=	\
 	--enable-alsa	\
 	--enable-audio
