diff --git a/package/ffmpeg/ffmpeg.mk b/package/ffmpeg/ffmpeg.mk
index 5d2f89f..3ac204f 100644
--- a/package/ffmpeg/ffmpeg.mk
+++ b/package/ffmpeg/ffmpeg.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-FFMPEG_VERSION = 3.4.2
+FFMPEG_VERSION = 3.3.5
 FFMPEG_SOURCE = ffmpeg-$(FFMPEG_VERSION).tar.xz
 FFMPEG_SITE = http://ffmpeg.org/releases
 FFMPEG_INSTALL_STAGING = YES
@@ -53,11 +53,14 @@ FFMPEG_CONF_OPTS = \
 	--disable-libdc1394 \
 	--disable-libgsm \
 	--disable-libilbc \
+	--disable-libnut \
+	--disable-libopenjpeg \
+	--disable-libschroedinger \
 	--disable-libvo-amrwbenc \
 	--disable-symver \
 	--disable-doc
 
-FFMPEG_DEPENDENCIES += host-pkgconf
+FFMPEG_DEPENDENCIES += $(if $(BR2_PACKAGE_LIBICONV),libiconv) host-pkgconf
 
 ifeq ($(BR2_PACKAGE_FFMPEG_GPL),y)
 FFMPEG_CONF_OPTS += --enable-gpl
@@ -158,10 +161,7 @@ endif
 ifeq ($(BR2_PACKAGE_FFMPEG_INDEVS),y)
 FFMPEG_CONF_OPTS += --enable-indevs
 ifeq ($(BR2_PACKAGE_ALSA_LIB),y)
-FFMPEG_CONF_OPTS += --enable-alsa
 FFMPEG_DEPENDENCIES += alsa-lib
-else
-FFMPEG_CONF_OPTS += --disable-alsa
 endif
 else
 FFMPEG_CONF_OPTS += --disable-indevs
@@ -254,10 +254,10 @@ else
 FFMPEG_CONF_OPTS += --disable-vdpau
 endif
 
-ifeq ($(BR2_PACKAGE_RPI_USERLAND),y)
+ifeq ($(BR2_PACKAGE_RPI_FIRMWARE)$(BR2_PACKAGE_RPI_USERLAND),yy)
 FFMPEG_CONF_OPTS += --enable-mmal --enable-omx --enable-omx-rpi \
 	--extra-cflags=-I$(STAGING_DIR)/usr/include/IL
-FFMPEG_DEPENDENCIES += rpi-userland
+FFMPEG_DEPENDENCIES += rpi-firmware rpi-userland
 else
 FFMPEG_CONF_OPTS += --disable-mmal --disable-omx --disable-omx-rpi
 endif
@@ -344,13 +344,6 @@ else
 FFMPEG_CONF_OPTS += --disable-libwavpack
 endif
 
-ifeq ($(BR2_PACKAGE_LIBICONV),y)
-FFMPEG_CONF_OPTS += --enable-iconv
-FFMPEG_DEPENDENCIES += libiconv
-else
-FFMPEG_CONF_OPTS += --disable-iconv
-endif
-
 # ffmpeg freetype support require fenv.h which is only
 # available/working on glibc.
 # The microblaze variant doesn't provide the needed exceptions
@@ -368,13 +361,6 @@ else
 FFMPEG_CONF_OPTS += --disable-fontconfig
 endif
 
-ifeq ($(BR2_PACKAGE_OPENJPEG),y)
-FFMPEG_CONF_OPTS += --enable-libopenjpeg
-FFMPEG_DEPENDENCIES += openjpeg
-else
-FFMPEG_CONF_OPTS += --disable-libopenjpeg
-endif
-
 ifeq ($(BR2_PACKAGE_X264)$(BR2_PACKAGE_FFMPEG_GPL),yy)
 FFMPEG_CONF_OPTS += --enable-libx264
 FFMPEG_DEPENDENCIES += x264
@@ -390,10 +376,10 @@ FFMPEG_CONF_OPTS += --disable-libx265
 endif
 
 ifeq ($(BR2_X86_CPU_HAS_MMX),y)
-FFMPEG_CONF_OPTS += --enable-x86asm
-FFMPEG_DEPENDENCIES += host-nasm
+FFMPEG_CONF_OPTS += --enable-yasm
+FFMPEG_DEPENDENCIES += host-yasm
 else
-FFMPEG_CONF_OPTS += --disable-x86asm
+FFMPEG_CONF_OPTS += --disable-yasm
 FFMPEG_CONF_OPTS += --disable-mmx
 endif
 
