--- a/configure.ac	2018-02-05 07:49:20.000000000 +0100
+++ b/configure.ac	2018-04-12 14:13:10.607698346 +0200
@@ -779,7 +779,7 @@
 if test $ac_lib_freetype = auto; then
 	AC_PATH_PROGS(
 		[ac_lib_freetype_config],
-		[${ac_tool_prefix}freetype-config],
+		[freetype-config],
 		[],
 		[${ac_lib_freetype_path}]
 	)
@@ -814,7 +814,7 @@
 elif test $ac_lib_freetype = yes; then
 	AC_PATH_PROGS(
 		[ac_lib_freetype_config],
-		[${ac_tool_prefix}freetype-config],
+		[freetype-config],
 		[],
 		[${ac_lib_freetype_path}]
 	)
@@ -890,7 +890,7 @@
 	else
 		AC_PATH_PROGS(
 			[ac_lib_sdl2_config],
-			[${ac_tool_prefix}sdl2-config],
+			[sdl2-config],
 			[],
 			[${ac_lib_sdl2_path}]
 		)
@@ -922,7 +922,7 @@
 elif test $ac_lib_sdl2 = yes; then
 	AC_PATH_PROGS(
 		[ac_lib_sdl2_config],
-		[${ac_tool_prefix}sdl2-config],
+		[sdl2-config],
 		[],
 		[${ac_lib_sdl2_path}]
 	)
