diff --git a/package/sdl2/sdl2.mk b/package/sdl2/sdl2.mk
index 3e3ba54aaf..2a95d39c6f 100644
--- a/package/sdl2/sdl2.mk
+++ b/package/sdl2/sdl2.mk
@@ -18,8 +18,7 @@ SDL2_CONF_OPTS += \
 	--disable-esd \
 	--disable-dbus \
 	--disable-pulseaudio \
-	--disable-video-wayland \
-	--disable-video-rpi
+	--disable-video-wayland
 
 # We must enable static build to get compilation successful.
 SDL2_CONF_OPTS += --enable-static
@@ -133,4 +132,12 @@ else
 SDL2_CONF_OPTS += --disable-video-kmsdrm
 endif
 
+ifeq ($(BR2_PACKAGE_RPI_FIRMWARE)$(BR2_PACKAGE_RPI_USERLAND),yy)
+	SDL2_DEPENDENCIES += rpi-firmware rpi-userland
+	SDL2_CONF_ENV += LDFLAGS="-L$(STAGING_DIR)/usr/lib -lvchostif"
+	SDL2_CONF_OPTS += --enable-video-rpi --enable-video-opengles
+else
+	SDL2_CONF_OPTS += --disable-video-rpi
+endif
+
 $(eval $(autotools-package))
