diff --git a/Source/Core/VideoBackends/OGL/RasterFont.cpp b/Source/Core/VideoBackends/OGL/RasterFont.cpp
index 8bc1b3d..48d3709 100644
--- a/Source/Core/VideoBackends/OGL/RasterFont.cpp
+++ b/Source/Core/VideoBackends/OGL/RasterFont.cpp
@@ -15,7 +15,6 @@
 namespace OGL
 {
 
-static const int CHAR_WIDTH = 8;
 static const int CHAR_HEIGHT = 13;
 static const int CHAR_OFFSET = 32;
 static const int CHAR_COUNT = 95;
