diff --git a/src/uinput.py b/src/uinput.py
index 8ab8d25..befb515 100644
--- a/src/uinput.py
+++ b/src/uinput.py
@@ -36,8 +36,8 @@ from steamcontroller.tools import get_so_extensions
 from collections import deque
 
 # Get All defines from linux headers
-if os.path.exists('/usr/include/linux/input-event-codes.h'):
-    CHEAD = defines('/usr/include', 'linux/input-event-codes.h')
+if os.path.exists('/usr/inc/input-event-codes.h'):
+    CHEAD = defines('/usr/inc', 'input-event-codes.h')
 else:
     CHEAD = defines('/usr/include', 'linux/input.h')
 
