diff --git a/package/kodi/kodi.mk b/package/kodi/kodi.mk
index e0361dae78..88072023a2 100644
--- a/package/kodi/kodi.mk
+++ b/package/kodi/kodi.mk
@@ -59,6 +59,7 @@ KODI_DEPENDENCIES = \
 	yajl \
 	zlib
 
+KODI_DEPENDENCIES += sdl sdl2
 KODI_CONF_ENV = \
 	PYTHON_VERSION="$(PYTHON_VERSION_MAJOR)" \
 	PYTHON_LDFLAGS="-lpython$(PYTHON_VERSION_MAJOR) -lpthread -ldl -lutil -lm" \
@@ -71,7 +72,9 @@ KODI_CONF_ENV = \
 
 KODI_CONF_OPTS +=  \
 	--with-ffmpeg=shared \
-	--disable-joystick \
+	--disable-crystalhd \
+	--disable-hal \
+	--enable-joystick \
 	--disable-openmax \
 	--disable-pulse \
 	--disable-vtbdecoder \
@@ -225,7 +228,7 @@ KODI_CONF_OPTS += --disable-libcec
 endif
 
 ifeq ($(BR2_PACKAGE_KODI_LIRC),y)
-KODI_CONF_OPTS += --enable-lirc
+KODI_CONF_OPTS += --with-lirc-device=/var/run/lirc/lircd
 else
 KODI_CONF_OPTS += --disable-lirc
 endif
