diff --git a/Makefile b/Makefile
index f066d1c..a8ce7c6 100644
--- a/Makefile
+++ b/Makefile
@@ -76,6 +76,15 @@ ifneq (,$(findstring unix,$(platform)))
         endif
     endif
 
+# ARM
+else ifneq (,$(findstring armv,$(platform)))
+    AR = ${CC_PREFIX}ar
+    CC = ${CC_PREFIX}gcc
+
+    TARGET := $(TARGET_NAME)_libretro.so
+    fpic := -fPIC
+    SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+
 else ifeq ($(platform), osx)
    TARGET := $(TARGET_NAME)_libretro.dylib
    fpic := -fPIC
