diff --git a/package/omxplayer/omxplayer.mk b/package/omxplayer/omxplayer.mk
index da4614bf56..69d76f9d4c 100644
--- a/package/omxplayer/omxplayer.mk
+++ b/package/omxplayer/omxplayer.mk
@@ -40,7 +40,13 @@ define OMXPLAYER_BUILD_CMDS
 endef
 
 define OMXPLAYER_INSTALL_TARGET_CMDS
-	$(INSTALL) -m 0755 -D $(@D)/omxplayer.bin $(TARGET_DIR)/usr/bin/omxplayer
+	$(INSTALL) -m 0755 -D $(@D)/omxplayer.bin $(TARGET_DIR)/usr/bin/omxplayer.bin
+        $(INSTALL) -m 755 $(@D)/omxplayer $(TARGET_DIR)/usr/bin/omxplayer
+        mkdir -p $(TARGET_DIR)/usr/omxplayer/
+        $(INSTALL) -m 755 $(@D)/dbuscontrol.sh $(TARGET_DIR)/usr/omxplayer/dbuscontrol.sh
+        mkdir -p $(TARGET_DIR)/usr/share/fonts/truetype/freefont/
+        $(INSTALL) -m 644 $(@D)/fonts/FreeSans.ttf $(TARGET_DIR)/usr/share/fonts/truetype/freefont/
+        $(INSTALL) -m 644 $(@D)/fonts/FreeSansOblique.ttf $(TARGET_DIR)/usr/share/fonts/truetype/freefont/
 endef
 
 $(eval $(generic-package))
