--- a/Makefile
--- b/Makefile
@@ -454,6 +454,7 @@
 CFLAGS += -g $(shell PKG_CONFIG_PATH=/usr/$(BASELIBDIR)/pkgconfig pkg-config $(SDL_LIB) --cflags)
 LFLAGS += -lm -L/usr/$(BASELIBDIR) $(shell PKG_CONFIG_PATH=/usr/$(BASELIBDIR)/pkgconfig pkg-config $(SDL_LIB) --libs) #-lX11
 CUSTOMOBJS = gui_x11.o
+EXTRAOBJS = oric_ch376_plugin.o ch376.o
 FILEREQ_OBJ = filereq_sdl.o
 MSGBOX_OBJ = msgbox_sdl.o
 TARGET = oricutron
