--- a/machine.c
+++ b/machine.c
@@ -965,7 +965,7 @@
   oric->prclose = 0;
   oric->lasttapefile[0] = 0;
   oric->keymap = KMAP_QWERTY;
-  oric->statusbar_mode = STATUSBARMODE_FULL;
+  oric->statusbar_mode = STATUSBARMODE_NONE;
   oric->popupstr[0] = 0;
   oric->newpopupstr = SDL_FALSE;
   oric->popuptime = 0;
@@ -1282,7 +1282,7 @@
              break;
            }
 
-           sprintf( vidcapname, "Capturing to video%02d.avi", vidcapcount );
+           sprintf( vidcapname, "Capturing to /recalbox/share/screenshots/video%02d.avi", vidcapcount );
            warpspeed = SDL_FALSE;
            ay_lockaudio( &oric->ay );
            vidcap = avi_open( &vidcapname[13], oricpalette, soundavailable&&soundon, oric->vid_freq );
