--- a/Makefile	2017-10-24 03:24:34.000000000 +0200
+++ b/Makefile	2017-11-13 21:16:43.247293813 +0100
@@ -58,6 +58,26 @@
 	SHARED := -shared -Wl,--version-script=link.T
 	CFLAGS += -fno-builtin -fno-exceptions -ffunction-sections
 	LIBM :=
+# ARM
+else ifneq (,$(findstring armv,$(platform)))
+       TARGET := $(TARGET_NAME)_libretro.so
+       fpic := -fPIC
+       SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+       CC = gcc
+       CXX = g++
+       PLATFORM_DEFINES += -marm
+       ifneq (,$(findstring softfloat,$(platform)))
+               PLATFORM_DEFINES += -mfloat-abi=softfp
+       else ifneq (,$(findstring hardfloat,$(platform)))
+               PLATFORM_DEFINES += -mfloat-abi=hard
+       endif
+       ifneq (,$(findstring neon,$(platform)))
+               FLAGS += -mfpu=neon
+               ASFLAGS += -mfpu=neon
+               HAVE_NEON = 1
+       endif
+       PLATFORM_DEFINES += -DARM
+
 else ifeq ($(platform), osx)
 	TARGET := $(TARGET_NAME)_libretro.dylib
 	fpic := -fPIC
