--- a/Makefile	2019-07-12 15:54:42.424872304 +0200
+++ b/Makefile	2019-07-12 15:53:37.460267465 +0200
@@ -324,13 +324,15 @@
 
 $(OBJ)/%.1bpp: %.png
 	-@$(MKDIR) -p $(dir $@)
-	rgbgfx -d 1 -h -o $@ $<
+	$(RGBDS_DIR)/rgbgfx -d 1 -h -o $@ $<
 
 $(OBJ)/BootROMs/SameBoyLogo.rle: $(OBJ)/BootROMs/SameBoyLogo.1bpp build/logo-compress
 	./build/logo-compress < $< > $@
 
+logo-compress: build/logo-compress
+
 build/logo-compress: BootROMs/logo-compress.c
-	$(CC) $< -o $@
+	$(HOST_CC) $< -o $@
 
 $(BIN)/BootROMs/agb_boot.bin: BootROMs/cgb_boot.asm
 $(BIN)/BootROMs/cgb_boot_fast.bin: BootROMs/cgb_boot.asm
@@ -338,8 +340,8 @@
 
 $(BIN)/BootROMs/%.bin: BootROMs/%.asm $(OBJ)/BootROMs/SameBoyLogo.rle
 	-@$(MKDIR) -p $(dir $@)
-	rgbasm -i $(OBJ)/BootROMs/ -i BootROMs/ -o $@.tmp $<
-	rgblink -o $@.tmp2 $@.tmp
+	$(RGBDS_DIR)/rgbasm -i $(OBJ)/BootROMs/ -i BootROMs/ -o $@.tmp $<
+	$(RGBDS_DIR)/rgblink -o $@.tmp2 $@.tmp
 	dd if=$@.tmp2 of=$@ count=1 bs=$(if $(findstring dmg,$@)$(findstring sgb,$@),256,2304)
 	@rm $@.tmp $@.tmp2
 
