--- a/Makefile	2017-10-24 19:32:07.000000000 +0200
+++ b/Makefile	2017-11-13 18:06:08.969538247 +0100
@@ -57,6 +57,25 @@
    OSX_LT_MAVERICKS = `(( $(OSXVER) <= 9)) && echo "YES"`
    fpic += -mmacosx-version-min=10.1
 
+# ARM
+else ifneq (,$(findstring armv,$(platform)))
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC
+   SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+   CC = gcc
+   CXX = g++
+   FLAGS += -marm
+      ifneq (,$(findstring softfloat,$(platform)))
+         FLAGS += -mfloat-abi=softfp
+      else ifneq (,$(findstring hardfloat,$(platform)))
+         FLAGS += -mfloat-abi=hard
+      endif
+      ifneq (,$(findstring neon,$(platform)))
+         FLAGS += -mfpu=neon
+         HAVE_NEON = 1
+      endif
+      FLAGS += -DARM
+
 # iOS
 else ifneq (,$(findstring ios,$(platform)))
 
