diff --git a/package/lirc-tools/lirc-tools.mk b/package/lirc-tools/lirc-tools.mk
index a5165ad..9800a4a 100644
--- a/package/lirc-tools/lirc-tools.mk
+++ b/package/lirc-tools/lirc-tools.mk
@@ -16,6 +16,7 @@ LIRC_TOOLS_AUTORECONF = YES
 
 LIRC_TOOLS_CONF_ENV = XSLTPROC=yes HAVE_WORKING_POLL=yes
 LIRC_TOOLS_CONF_OPTS = --without-x --enable-devinput --enable-uinput
+LIRC_TOOLS_MAKE=$(MAKE1)
 
 ifeq ($(BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_4),y)
 LIRC_TOOLS_CONF_ENV += \
