diff --git a/Makefile.libretro b/Makefile.libretro
index 26ee1c1..ae4a2aa 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -377,6 +377,14 @@ else ifeq ($(platform), qnx)
 	use_sh2drc = 1
 	use_svpdrc = 1
 
+# AARCH64 generic
+else ifeq ($(platform), aarch64)
+	TARGET := $(TARGET_NAME)_libretro.so
+	ARCH = aarch64
+   fpic := -fPIC
+	SHARED := -shared
+	DONT_COMPILE_IN_ZLIB = 1
+	CFLAGS += -DFAMEC_NO_GOTOS
 # ARM
 else ifneq (,$(findstring armv,$(platform)))
 	TARGET := $(TARGET_NAME)_libretro.so
