diff --git a/package/kodi-visualisation-fishbmc/Config.in b/package/kodi-visualisation-fishbmc/Config.in
index 9e2eabbdbc..be638afbbc 100644
--- a/package/kodi-visualisation-fishbmc/Config.in
+++ b/package/kodi-visualisation-fishbmc/Config.in
@@ -1,5 +1,6 @@
 config BR2_PACKAGE_KODI_VISUALISATION_FISHBMC
 	bool "kodi-visualisation-fishbmc"
+    depends on BR2_PACKAGE_KODI
 	depends on BR2_PACKAGE_HAS_LIBGL
 	help
 	  Fische visualiser for Kodi
