diff --git a/configure b/configure
index 5b069eb..8c1118b 100755
--- a/configure
+++ b/configure
@@ -5780,7 +5780,7 @@ enabled opengl            && { check_lib GL/glx.h glXGetProcAddress "-lGL" ||
 enabled omx_rpi && enable omx
 enabled omx               && { check_header OMX_Core.h ||
                                 { ! enabled cross_compile && enabled omx_rpi && {
-                                    add_cflags -isystem/opt/vc/include/IL ; }
+                                    add_cflags -isystem/usr/include/IL ; }
                                 check_header OMX_Core.h ; } ||
                                die "ERROR: OpenMAX IL headers not found"; }
 enabled openssl           && { use_pkg_config openssl openssl/ssl.h SSL_library_init ||
