diff --git a/libsrc/leddevice/LedDevicePhilipsHue.cpp b/libsrc/leddevice/LedDevicePhilipsHue.cpp
index fe9e359..5838534 100755
--- a/libsrc/leddevice/LedDevicePhilipsHue.cpp
+++ b/libsrc/leddevice/LedDevicePhilipsHue.cpp
@@ -105,10 +105,10 @@ CiColor PhilipsHueLight::rgbToCiColor(float red, float green, float blue) {
 	// Convert to x,y space.
 	float cx = X / (X + Y + Z);
 	float cy = Y / (X + Y + Z);
-	if (std::isnan(cx)) {
+	if (isnan(cx)) {
 		cx = 0.0f;
 	}
-	if (std::isnan(cy)) {
+	if (isnan(cy)) {
 		cy = 0.0f;
 	}
 	// Brightness is simply Y in the XYZ space.
